File: MusicWeb.Server/MusicWeb.Models/Entities/Posts/Post.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Entities.Posts.Base;
using MusicWeb.Models.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities.Posts
{
    public class Post : BaseEntity
    {
        public string Text { get; set; }
        public DateTime CreateDate { get; set; }

        public string PosterId { get; set; }
        public int? ArtistPosterId { get; set; }
        public int? AlbumId { get; set; }

        public virtual Artist PosterArtist { get; set; }
        public virtual Album Album { get; set; }
        public virtual UserFriend Poster { get; set; }
        public virtual ICollection<PostLike> PostLikes { get; set; }
        public virtual ICollection<PostComment> PostComments { get; set; }
    }
}

File: MusicWeb.Client/src/models/Post.js
export default class Post {
  userName = "";
  artistId = null;
  artist = {};
  album = {};
  id = null;
  text = "";
  createDate = "";
  posterId = null;
  artistPosterId = "";
  albumId = "";
  image = "";
  isLiked = false;
  totalLikes = null;
  albumImage = null;

  constructor($data) {
    if ($data) {
      this.userName = $data.userName;
      this.artistId = $data.artistId;
      this.artist = $data.artist;
      this.album = $data.album;
      this.id = $data.id;
      this.text = $data.text;
      this.createDate = $data.createDate;
      this.posterId = $data.posterId;
      this.artistPosterId = $data.artistPosterId;
      this.albumId = $data.albumId;
      this.image = $data.image;
      this.isLiked = $data.isLiked;
      this.totalLikes = $data.totalLikes;
      this.albumImage = $data.albumImage;
    }
  }
}
